sw/tools/opti_dist/dist.py

```diff
@@ -0,0 +1,66 @@
+﻿import time
+import sys
+import math
+import matplotlib.pyplot as plt
+from NatNetClient import NatNetClient
+
+pos_x, pos_y, pos_z = 0.0, 0.0, 0.0
+plt.axis([-6, 6, -6, 6])
+
+# This is a callback function that gets connected to the NatNet client and called once per mocap frame.
+def receiveNewFrame( frameNumber, markerSetCount, unlabeledMarkersCount, rigidBodyCount, skeletonCount,
+                    labeledMarkerCount, latency, timecode, timecodeSub, timestamp, isRecording, trackedModelsChanged ):
+    # print( "Received frame", frameNumber )
+	pass
+
+# This is a callback function that gets connected to the NatNet client. It is called once per rigid body per frame
+def receiveRigidBodyFrame( id, position, rotation ):
+    # print( "Received frame for rigid body", id )
+	if id==13:
+		global pos_x, pos_y, pos_z
+		pos_x = position[0]
+		pos_y = position[1]
+		pos_z = position[2]
+		# precision = 1
+		# pos_x = round(pos_x, precision)
+		# pos_y = round(pos_y, precision)
+		# pos_z = round(pos_z, precision)
+	else:
+		pass
+
+# This will create a new NatNet client
+streamingClient = NatNetClient()
+# Configure the streaming client to call our rigid body handler on the emulator to send data out.
+streamingClient.newFrameListener = receiveNewFrame
+streamingClient.rigidBodyListener = receiveRigidBodyFrame
+# Start up the streaming client now that the callbacks are set up.
+# This will run perpetually, and operate on a separate thread.
+streamingClient.run()
+
+time.sleep(2)
+print('Start tracking')
+# file = open('opti.csv', 'w')

```

Comment:
I suggest to use runtime argument to specify the log file name and write the data when needed rather than editing this file.

Reply:
I have cleaned up the script and added the output file as an argument. Will re-test asap.